install.sh

```diff
@@ -178,25 +244,239 @@ legacy_transformations_end() {
+            echo "  hook repositories, you should update all shared hook repositories" >&2
+            echo "  by running in all repositories using Githooks:"
+            echo "    \$ git hooks shared update"
+            echo "  The Git config variable \`githooks.failOnNonExistingSharedHooks\` has been" >&2
+            echo "  enabled globally to safely fail if you forgot to update them." >&2
+            echo >&2
+            git config --global githooks.failOnNonExistingSharedHooks "true"
+
+        fi
+    else
+
+        LIST="$INSTALL_DIR/registered"
+        if [ -f "$LIST" ]; then
+            IFS="$IFS_NEWLINE"
+            while read -r REGISTERED_REPO; do
+                unset IFS
+
+                if [ "$(git -C "$REGISTERED_REPO" rev-parse --is-inside-git-dir 2>/dev/null)" = "false" ]; then
+                    # Not existing git dir -> skip.
+                    true
+                else
+                    # Remove legacy values
+                    git -C "$REGISTERED_REPO" config --local --unset githooks.autoupdate.registered
+
+                    if [ "$PR_125" = "true" ]; then
+
+                        legacy_transform_adjust_local_paths "$REGISTERED_REPO" ||
+                            LEGACY_TRANSFORM_FAILURES="true"
+
+                        legacy_transform_update_shared_hooks "$REGISTERED_REPO" ||
+                            LEGACY_TRANSFORM_FAILURES="true"
+                    fi
+                fi
+
+                IFS="$IFS_NEWLINE"
+            done <"$LIST"
+        fi
+
+    fi
+
+    return 0
+}
+
+############################################################
+# Function to adjust local paths in `.githooks/.shared`
+#   file which are forbidden.
 #
 # Returns:
 #   1 when failed, 0 otherwise
 ############################################################
-legacy_transformations_repo_end() {
+legacy_transform_adjust_local_paths() {
+
+    if [ "$(git -C "$1" rev-parse --is-bare-repository 2>/dev/null)" = "true" ]; then
+        SHARED_FILE="$1/.githooks/.shared"
+    else
+        SHARED_FILE="$1/../.githooks/.shared"
+    fi
+
+    if [ -f "$SHARED_FILE" ]; then
+
+        NEW_SHARED_LIST=$(mktemp)
+        MOVED_URLS=$(mktemp)
+
+        MOVED="false"
+        IFS="$IFS_NEWLINE"
+        while read -r LINE || [ -n "$LINE" ]; do
+            unset IFS
+
+            if echo "$LINE" | grep -qE "^\s*(#.*)?$"; then
+
+                echo "$LINE" >>"$NEW_SHARED_LIST"
 
-    git -C "$1" config --local --unset githooks.autoupdate.registered
+            elif is_local_path "$LINE" || is_local_url "$LINE"; then
+                git -C "$1" config --local --add githooks.shared "$LINE"
+
+                echo "$LINE" >>"$MOVED_URLS"
+                MOVED="true"
+            else
+                echo "$LINE" >>"$NEW_SHARED_LIST"
+            fi
+
+            IFS="$IFS_NEWLINE"
+        done <"$SHARED_FILE"
+
+        cp -f "$NEW_SHARED_LIST" "$SHARED_FILE" &&
```

Comment:
I'd only do this inside the `if [ "$MOVED" = "true" ]` block, because a no change might still show up as a git change if permissions change, etc.

Reply:
On that note, the warning below should probably tell the user that this file has changed and they'll need to commit the changes.

Reply:
Good point!